File: Assignments/Assignment_27/p8.c
#include<stdio.h>

int isArmstrong(int num){

        int temp=num,count=0,sum=0,dig;
        while(temp>0){

                count++;
                temp/=10;
        }temp=num;
        while(temp>0){

                dig=temp%10;
                int num1=dig;
                for(int i=0;i<count-1;i++){

                        num1*=dig;
                }sum+=num1;
                temp/=10;
        }if(num==sum){

                return 1;
        }else{return 0;}
}

void main(){

        int num;
        printf("Enter a number:\n");
        scanf("%d",&num);
        if(isArmstrong(num)){

                printf("%d is an armstrong number\n",num);
        }else{printf("%d is not an armstrong number\n",num);}

}

File: JAVA/14.Number_System/p8.java
import java.util.*;

class Demo{

	public static void main(String[] args){
	
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter a number:");
		int num=sc.nextInt();
		int temp=num,dig=0,sum=0;
		int size=0,index=0;
		while(temp>0){
		
			dig=temp%10;
			if(dig%2!=0){
			
				System.out.println(dig*dig);
			}else{
			
				System.out.println(dig);
			}
			temp/=10;
		}
	}
}
